.github/workflows/autobuild.yml

```diff
@@ -251,77 +233,81 @@ jobs:
 
       - name:                       "Post-Build for ${{ matrix.config.config_name }}"
         id:                         step_cmd3_postbuild
-        if:                         ${{ matrix.config.cmd3_postbuild }}
+        if:                         matrix.config.cmd3_postbuild
         run:                        ${{ matrix.config.cmd3_postbuild }} ${{ github.workspace }}
         env:
           jamulus_project_path:       ${{ github.workspace }}
           jamulus_buildversionstring: ${{ needs.create_release.outputs.version_name }}
 
-      # Upload Artifact to Job
       - name:                       Upload Artifact 1 to Job
-        if:                         ${{ steps.step_cmd3_postbuild.outputs.artifact_1 }}
+        if:                         steps.step_cmd3_postbuild.outputs.artifact_1
         uses:                       actions/upload-artifact@v2
         with:
           name:                     ${{ steps.step_cmd3_postbuild.outputs.artifact_1 }}
           path:                     deploy/${{ steps.step_cmd3_postbuild.outputs.artifact_1 }}
           retention-days:           31
-          if-no-files-found:        error # 'warn' or 'ignore' are also available, defaults to `warn`
+          if-no-files-found:        error
 
-      # Upload Artifact to Job
       - name:                       Upload Artifact 2 to Job
-        if:                         ${{ steps.step_cmd3_postbuild.outputs.artifact_2 }}
+        if:                         steps.step_cmd3_postbuild.outputs.artifact_2
         uses:                       actions/upload-artifact@v2
         with:
           name:                     ${{ steps.step_cmd3_postbuild.outputs.artifact_2 }}
           path:                     deploy/${{ steps.step_cmd3_postbuild.outputs.artifact_2 }}
           retention-days:           31
-          if-no-files-found:        error # 'warn' or 'ignore' are also available, defaults to `warn`
-
-      # Notarize macOS build, if needed
-      - name: "Notarize macOS Release Build"
-        if: ${{ (steps.step_cmd3_postbuild.outputs.artifact_1 != '') && (steps.step_macos_build.outputs.macos_signed == 'true') && contains(needs.create_release.outputs.publish_to_release, 'true') }}
-        id: notarize-macOS-app
-        uses: devbotsxyz/xcode-notarize@d7219e1c390b47db8bab0f6b4fc1e3b7943e4b3b
+          if-no-files-found:        error
+
+      - name:                       "Notarize macOS Release Build"
+        if:                         >-
+                                    steps.step_cmd3_postbuild.outputs.artifact_1 != '' &&
+                                    steps.step_macos_build.outputs.macos_signed == 'true' &&
+                                    contains(needs.create_release.outputs.publish_to_release, 'true')
+        id:                         notarize-macOS-app
+        uses:                       devbotsxyz/xcode-notarize@d7219e1c390b47db8bab0f6b4fc1e3b7943e4b3b
```

Comment:
Why is this done after the upload artifact call? Couldn't there also be signed non-release artifacts?

Reply:
Hm, currently it only runs on release builds. It could probably also be run earlier. Not sure if there are any reasons for not signing dev builds, but as this runs on @emlynmac's fork and not on ours anyway, I'm not sure how deeply we should care...? :)

Maybe open a separate issue for that?

Reply:
@emlynmac could you please comment why this is here and not before any upload has started?

Reply:
This step is to notarize the build, which has to happen after the build is done and the artifacts created.
The signing of the macOS build is a two-step process; build with certs then notarize the full thing with Apple.

Reply:
Is this basically a step which transfers something (hashes?) to apple?

Reply:
There's a bunch of information about how notarization works here: https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution

Reply:
@hoffie sorry for the small modification request: Could you please add a comment here that this needs to be done after the build is created?

Reply:
Not sure I understand.
Do you mean that the binaries can only be signed once they're built? That would be rather obvious to me (similar for other instances: We don't explicitly highlight that the build dependencies have to be prepared before the build can start).

Or do you mean that the binaries can only be signed once the artifact has been uploaded to the job? I don't think this is the case (or if it is, I don't understand why). The Notarize step references `product-path:             deploy/${{ steps.step_cmd3_postbuild.outputs.artifact_1 }}`. This looks like a filesystem path to me and not like a job artifact path. So I guess the order could be turned around. Adding a comment which states the contrary would seem strange then :)

Maybe you could send a follow-up PR with a suggested wording?

(In general, I think a PR which touches specific code should leave the code in better state than it was before. However, we should be cautious to avoid trying to push to much *additional* things which *could* or *should* be improved into PRs. PRs are getting large and take lots of iterations until they can get merged...)

Reply:
Yes sure. @emlynmac should open a PR then since he knows what to do. I think we can move it around - at least from my readings.